forced_directives_map.go

```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package crossplane
+
+// A human editable map. Used to overwrite generated directive definitions for special cases.
```

Comment:
It's unclear to me what "forcedMap" means. I think there might be a better name for this variable. I also think that the comment isn't adding clarity. I'm not sure why `if` is a special case necessitating it be put into it's own map. 

All code is by default "editable". It's a common Go convention to add a comment at the top of the file containing generated code that should not be edited directly, something like:

```go
// Code generated by thinger -option Thingy thing.go; DO NOT EDIT.
```

Reply:
> I'm not sure why `if` is a special case necessitating it be put into its own map.

The reason is: `if` has a  "crossplane created" bitmask `ngxConfExpr`. `ngxConfExpr` was not in source code and crossplane created it to indicate a directive followed by expression in parentheses `()`. I may caution there will be similar cases in the future so I want to provide a way for human to override the generated map

> It's a common Go convention to add a comment at the top of the file containing generated code that should not be edited directly, something like:

I had `// Code generated by generator; DO NOT EDIT.` in these files but not at the top. I think copyright should appear first?(I failed to find an example, chatGPT said copyright first)

> I think there might be a better name for this variable

Sure.... I struggled a lot for it. Just thought a name `priorityDirectiveMap`. Maybe it's better?